podcast/37/transcript.markdown

```diff
@@ -0,0 +1,253 @@
+
+And so, it really became difficult to use filters. They had to have a lot of Haskell stuff installed and they had to have it installed properly and they had to have libraries set up right. So then I thought, well, what if we just build a Lua interpreter into it, then they don’t need any external interpreter at all. And there’s other advantages to that as well. So, one advantage is you skip all the serialization and deserialization, which adds a lot of time to what it takes to use a filter. So, with the Lua filters, the Lua interpreter is directly hooked into the Pandoc code, so it can just go in and mess around with the types and things directly. There’s very little overhead to running a Lua filter, and you need no external software. So, it’s huge gains, I think, in usability. It’s true you don’t have the types, and it can make it a little more difficult to debug, but I think overall it’s been a big win. And there are lots of projects now that are using just tons of Lua filters on Pandoc. So, I think R Markdown, which is used for the R statistics language, is one. It’s just a bunch of, as I understand it, Lua filters. I hope that’s – that might be inaccurate actually. So, maybe I’m getting confused between R Markdown and another project, but there’s a philosophy journal now that runs with Pandocs and a whole mess of Lua filters. So, it’s turned out to be a pretty practical decision.
+
+*JB (0:18:38)*: Are you now running into problems where previously you could change the AST easily, and because if you have the types in the refactoring and *(0:18:46 unintelligible)* Haskell, you just change everything and it works again, and now you have to worry about all these untied Lua filters out there that you maybe don’t want to break anymore? 
```

Comment:
```suggestion
*JB (0:18:38)*: Are you now running into problems where previously you could change the AST easily, and because if you have the types in the refactoring and it’s great in  Haskell, you just change everything and it works again, and now you have to worry about all these untied Lua filters out there that you maybe don’t want to break anymore? 
```